### PR TITLE
Solve Sprockets DEPRECATION WARNING

### PR DESCRIPTION
**Pull request by brendon**
I’m using Grease https://github.com/yasaichi/grease as a wrapper around
`LessTemplate` and `ImportProcessor` and have also rejigged the loading
process to make things compatible with Sprockets 2, 3 and 4.

Extra changes:
* Turned the warnings off for now with rake to make the output of the
tests a bit more understandable.
* Update the `README` with regards to getting the test suite running
* Fixed a couple of spec’s to look for the right engine and
preprocessor classes
* Removed the use of `capture` in `run_generator`. There is no
recommended replacement.

Closes: #122 Finally! :)

Thanks to https://github.com/metaskills/less-rails/pull/128 for inspiration @MustafaZain.

### CODE REVIEW

**Comment by brendon** [comment]
@simi, finally the test suite seems to be in order! Let me know if you need any more information about this. I think there's a large group of people hanging out for this fix :D

**Comment by tdonia** [comment]
this works for us (thank you, @brendon!) -- almost -- on Rails 5.0 because it eliminates `Sprockets.register_engine '.less', LessTemplate` in railties in this commit: https://github.com/brendon/less-rails/commit/f573927f2de15b611f797fb650cb95856ec71d95#diff-998c29126ad83cee99c229f1d43307b5L16

the one additional issue that we hit is the line above:

`Sprockets::Engines #force autoloading` errs with
`block in <class:Railtie>': uninitialized constant Sprockets::Engines (NameError)`

**Comment by brendon** [comment]
Thanks for the feedback @tdonia :) I've pushed a change that should hopefully fix that. Let me know how it works on your end. I'll see how travis likes it also.

**Comment by ali-anwar** [comment]
Any update on this? Thank you.

File: README.md [hunk 1]
```diff
@@ -128,9 +128,9 @@ We also have a generator for rails scaffold CSS. Just like the Sass gem, we simp
 This gem is fully tested from Rails 4.0 to Rails 4.2. We run our tests on [Travis CI](http://travis-ci.org/metaskills/less-rails) in both Ruby 1.9, 2.0, and jRuby 1.9 mode. If you detect a problem, open up a github issue or fork the repo and help out. After you fork or clone the repository, the following commands will get you up and running on the test suite. 
 
 ```shell
-$ bundle install
-$ bundle exec rake appraisal:setup
-$ bundle exec rake appraisal test
+$ bundle
```
Comment (simi) [inline README.md]: please keep `bundle install` here

File: Rakefile [hunk 2]
```diff
@@ -7,6 +7,7 @@ Rake::TestTask.new do |t|
   t.libs = ['lib','test']
   t.test_files = Dir.glob("test/**/*_spec.rb").sort
   t.verbose = true
+  t.warning = false
```
Comment (simi) [inline Rakefile]: Can you remove this change?

File: .travis.yml [hunk 3]
```diff
@@ -1,14 +1,22 @@
+language: ruby
+cache: bundler
+sudo: false
 rvm:
-  - 1.9.3
-  - 2.0.0
-  - 2.1.4
-  - jruby-19mode
+  - 2.2.7
```
Comment (simi) [inline .travis.yml]: Do we need to drop old rubies here? Are they failing?

File: README.md (hunk 1 above)
Comment (brendon) [inline README.md]: I can do that, but `bundle` is the same as `bundle install` (install is the default action). Let me know if you still want it changed back.

File: Rakefile (hunk 2 above)
Comment (brendon) [inline Rakefile]: Will do. I think it was to silence the errors from some other dependencies (`therubyracer`) and make the test results for rational to view.

File: .travis.yml (hunk 3 above)
Comment (brendon) [inline .travis.yml]: Off the top of my head I can't remember. Depending on the rails version, we'd have to exclude certain combinations. I can add them back to the matrix and see how they test out if you like?

**Comment by brendon** [comment]
Hi @simi, I've updated the test matrix and all the tests pass. I'm only testing for the latest version of jruby. Is that ok?

I've commented out the warnings line. It's handy to have there locally to uncomment because of the excessive `therubyracer` warnings. We should migrate away from that perhaps in another PR since I think the Rails team did the same.

I'll wait to hear from you on the `bundle` vs `bundle install` thing.

**Comment by brendon** [comment]
Hi @simi, any more comments before we finalise this? :)

**Comment by drn** [comment]
@simi Can we get this merged in? We just upgrade sprockets / rails and are running into these warnings. Happy to help make any changes, but it looks like the changes have been made and we're just waiting on a merge + release.

**Comment by ubugnu** [comment]
I've used the workaround:
`gem 'less-rails', github: 'brendon/less-rails', branch: 'fix-sprockets-loading'`
But less files are no longer compiled (I get a 'not found' myscript.less.css)
The same configuration works with the default gem.
Any idea on how to fix the problem?

**Comment by brendon** [comment]
Hi @ubugnu, thanks for reporting the error. If `.less.css` is a usual extension then the problem is that it wasn't added as a valid extension for Sprockets 3 that requires all extensions to be explicitly defined.

I've added the `.less.css` extension in there for now. Give it a try. Can others chime in on whether `.less.css` is a usual extension for less files in Rails?

File: README.md (hunk 1 above)
Comment (darthrevan) [reply README.md]: Anyone alive?

Comment (brendon) [reply README.md]: I can confirm that I am alive.

**Comment by brendon** [comment]
100,000 🎉 :D

**Comment by brendon** [comment]
Can you ping when the gem is released? :)

**Comment by simi** [comment]
First of all I'm really sorry for be the break at this deprecation removal problem.

I would like to thanks all involved in this, mainly @brendon for crafting this PR and @drn for friendly email ping.

This is included in fresh 3.0 release.

**Pull request merged**